### PR TITLE
Feature/#59 external nodes

### PR DESCRIPTION
**Pull request by pbimpisidis**

### CODE REVIEW

File: it/ExternalNode.Tests.ps1 [hunk 1]
```diff
@@ -8,283 +10,263 @@ function Stop-Pester($message = "EMERGENCY: Script cannot continue.")
...
-
-	Context "ExternalNode_PinDown" {
 	
-		BeforeAll {
-            Import-Module biz.dfch.PS.Appclusive.Client
-			$svc = Enter-ApcServer;
+	$entityPrefix = "TestItem-";
+	$usedEntitySets = @("Nodes", "ExternalNodes", "ExternalNodeBags");
+	$nodeEntityKindId = [biz.dfch.CS.Appclusive.Public.Constants+EntityKindId]::Node.value__;
+	$nodeParentId = (Get-ApcTenant -Current).NodeId;
```
Comment (rufer7) [inline it/ExternalNode.Tests.ps1]: @pbimpisidis here you have to pass `$svc` to the `Get-ApcTenant` Cmdlet but `$svc` is not yet defined at this time so I suggest defining $nodeParentId in BeforeEach

File: it/ExternalNode.ps1 [hunk 2]
```diff
@@ -1,4 +1,126 @@
...
+	(
+		$Svc
+		,
+		$Name
+		,
+		$Description = "Test External Node Bag"
+		,
+		$Value = [biz.dfch.CS.Appclusive.Public.Constants+EntityKindId]::EntityBag.value__
+		,
+		[Parameter(Mandatory=$true)] $ExternalNodeId
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis please put `$ExternalNodeId` on a new line

File: it/ExternalNode.ps1 [hunk 3]
```diff
@@ -1,4 +1,126 @@
...
+function Create-ExternalNodeBag {
+	Param
+	(
+		$Svc
+		,
+		$Name
+		,
+		$Description = "Test External Node Bag"
+		,
+		$Value = [biz.dfch.CS.Appclusive.Public.Constants+EntityKindId]::EntityBag.value__
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis I suggest setting the default value as follows as it has not to be an entityKindId:

`$Value = "Arbitrary Value"`

File: it/ExternalNode.ps1 [hunk 4]
```diff
@@ -1,4 +1,126 @@
...
+		,
+		[Parameter(Mandatory=$true)] $ExternalNodeId
+		,
+		$Tid = (Get-ApcTenant -Current).Id
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis default value for `$Tid` has to be set as follows: `(Get-ApcTenant -Current -svc $svc).Id` but `$svc` is not yet set when setting the default value in parameters...

File: it/ExternalNode.ps1 [hunk 5]
```diff
@@ -1,4 +1,126 @@
...
+	)
+	
+	#add parameters
+	$newExternalNodeBag = New-Object biz.dfch.CS.Appclusive.Api.Core.ExternalNodeBag;
+	$newExternalNodeBag.Name = $name;
+	$newExternalNodeBag.Description = $description;
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis change to `$Description`

File: it/ExternalNode.ps1 [hunk 6]
```diff
@@ -1,4 +1,126 @@
...
+	$newExternalNodeBag.Value = $value;
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis change to `$Value`

File: it/ExternalNode.ps1 [hunk 7]
```diff
@@ -1,4 +1,126 @@
...
+	$newExternalNodeBag.ExternaldNodeId = $externalNodeId;
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis change to `$ExternalNodeId`

File: it/ExternalNode.ps1 [hunk 8]
```diff
@@ -1,4 +1,126 @@
...
+	$newExternalNodeBag.Tid = $tid;
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis change to `$Tid`

File: it/ExternalNode.ps1 [hunk 9]
```diff
@@ -1,4 +1,126 @@
...
+	
+	#ACT create external node bag
+	$svc.Core.AddToExternalNodeBags($newExternalNodeBag);
+	$result = $svc.Core.SaveChanges();
+	
+	#get external node bag
+	$query = "Id eq {0}" -f $newExternalNodeBag.Id;
+	$externalNodeBag = $svc.Core.ExternalNodeBags.AddQueryOption('$filter', $query) | select;
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis Select with uppercase at the beginning please

File: it/ExternalNode.ps1 [hunk 10]
```diff
@@ -1,4 +1,126 @@
...
+	
+	#ASSERT external node bag
+	$bin = $result.StatusCode | Should Be 201;
+	$bin = $externalNodeBag | Should BeOfType [biz.dfch.CS.Appclusive.Api.Core.ExternalNodeBag];
+	$bin = $externalNodeBag | Should Not Be $null;
+	$bin = $externalNodeBag.Name = $name;
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis change to `$Name`

File: it/ExternalNode.ps1 [hunk 11]
```diff
@@ -1,4 +1,126 @@
...
+	$bin = $externalNodeBag.Description = $description;
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis change to `$Description`

File: it/ExternalNode.ps1 [hunk 12]
```diff
@@ -1,4 +1,126 @@
...
+	$bin = $externalNodeBag.Value = $value;
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis change to `$Value`

File: it/ExternalNode.ps1 [hunk 13]
```diff
@@ -1,4 +1,126 @@
...
+	$bin = $externalNodeBag.ExternaldNodeid = $externalNodeId;
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis change to `$ExternalNodeId`

File: it/ExternalNode.ps1 [hunk 14]
```diff
@@ -1,4 +1,126 @@
...
+	$bin = $externalNodeBag.Tid = $tid;
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis change to `$Tid`

File: it/ExternalNode.ps1 [hunk 15]
```diff
@@ -1,4 +1,126 @@
...
+		,
+		$UpdatedDescription
+		,
+		$UpdatedExternalType
+		,
+		$UpdatedExternalId
+	)
+	
+	#get the external node
+	$externalNode = Get-ApcExternalNode -Id $externalNodeId;
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis change to `$ExternalNodeId`

File: it/ExternalNode.ps1 [hunk 16]
```diff
@@ -1,4 +1,126 @@
...
+	$externalNode.Name = $UpdatedName;
+	$externalNode.Description = $UpdatedDescription;
+	$externalNode.ExternalType = $UpdatedExternalType;
+	$externalNode.ExternalId = $UpdatedExternalId;
+	
+	$svc.Core.UpdateObject($externalNode);
+	$result = $svc.Core.SaveChanges();
+	
+	#get the updated external node
+	$updatedExternalNode = Get-ApcExternalNode -Id $externalNodeId;
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis change to `$ExternalNodeId`

File: it/ExternalNode.ps1 [hunk 17]
```diff
@@ -1,4 +1,126 @@
...
+	
+	#ASSERT - update
+	$bin = $updatedExternalNode.Id | Should Be $externalNodeId;
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis change to `$ExternalNodeId`

File: it/ExternalNode.ps1 [hunk 18]
```diff
@@ -1,4 +1,126 @@
...
+		,
+		$UpdatedName
+		,
+		$UpdatedDescription
+		,
+		$UpdatedValue
+	)
+	
+	#get the external node bag
+	$filter = "Id eq {0}" -f $externalNodeBagId;
```
Comment (rufer7) [inline it/ExternalNode.ps1]: @pbimpisidis change to `$ExternalNodeBagId`

File: it/ExternalNode.Tests.ps1 (hunk 1 above)
Comment (pbimpisidis) [inline it/ExternalNode.Tests.ps1]: fixed

File: it/ExternalNode.ps1 (hunk 2 above)
Comment (pbimpisidis) [inline it/ExternalNode.ps1]: done

File: it/ExternalNode.ps1 (hunk 3 above)
Comment (pbimpisidis) [inline it/ExternalNode.ps1]: fixed

File: it/ExternalNode.ps1 (hunk 5 above)
Comment (pbimpisidis) [inline it/ExternalNode.ps1]: fixed

File: it/ExternalNode.ps1 (hunk 6 above)
Comment (pbimpisidis) [inline it/ExternalNode.ps1]: done

**Pull request merged**